pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -837,7 +837,7 @@ func (l *LibvirtDomainManager) MigrateVMI(vmi *v1.VirtualMachineInstance, option
 }
 
 var updateHostsFile = func(entry string) (err error) {
-	file, err := os.OpenFile("/etc/hosts", os.O_WRONLY|os.O_APPEND, 0644)
+	file, err := os.OpenFile("/etc/hosts", os.O_WRONLY|os.O_APPEND, 0600)
```

Comment:
This is effectively a false positive, as the file should always be there, and in the "should never happen" case that it's not existing it will fail.

What the rule _should have done_ instead of flagging right away: evaluate the flags to be sure whether creating a file is even allowed. We should create an issue on the gosec repo for that.

Reply:
Thanks Daniel, this was a false positive, the mode is kept as 644.